lib/reek/source/source_code.rb

```diff
@@ -59,12 +59,13 @@ def initialize(code:, origin:, parser: self.class.default_parser)
       #
       # @return an instance of SourceCode
       # :reek:DuplicateMethodCall { max_calls: 2 }
-      def self.from(source)
+      # :reek:ControlParameter
+      def self.from(source, filename: nil)
         case source
         when File     then new(code: source.read,           origin: source.path)
-        when IO       then new(code: source.readlines.join, origin: IO_IDENTIFIER)
+        when IO       then new(code: source.readlines.join, origin: filename || IO_IDENTIFIER)
         when Pathname then new(code: source.read,           origin: source.to_s)
-        when String   then new(code: source,                origin: STRING_IDENTIFIER)
+        when String   then new(code: source,                origin: filename || STRING_IDENTIFIER)
```

Comment:
Why is `filename` used with both `IO` and `String`? For STDIN use I think only `IO` would be needed, so if we're going to add this to others, we might as well add this to all of them.

However, see my comment on `Examiner` about not passing filename to `SourceCode` at all.